File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/enums/MenuTypeEnum.java
package ink.ckx.mo.system.api.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import ink.ckx.mo.common.core.base.IBaseEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 菜单类型枚举
 *
 * @author chenkaixin
 */
@AllArgsConstructor
@Getter
public enum MenuTypeEnum implements IBaseEnum<Integer> {

    NULL(0, null),
    CATALOG(1, "目录"),
    MENU(2, "菜单"),
    BUTTON(3, "按钮");

    @JsonValue
    @EnumValue
    private final Integer value;

    private final String label;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/enums/MenuTypeEnum.kt
package ink.ckx.mo.admin.api.enums

import com.baomidou.mybatisplus.annotation.EnumValue
import com.fasterxml.jackson.annotation.JsonValue
import ink.ckx.mo.common.core.base.IBaseEnum

/**
 * 菜单类型枚举
 *
 * @author chenkaixin
 */
enum class MenuTypeEnum(

    @JsonValue
    @EnumValue
    override var value: Int,
    override var label: String
) : IBaseEnum<Int> {

    CATALOG(1, "目录"),
    MENU(2, "菜单"),
    BUTTON(3, "按钮"),
    ;
}